File: Assignment_42/Program4.c
/*
Write a recursive program which accept number from user and return its factorial
Input : 5
Output : 120
*/

#include <stdio.h>

int Fact(int iNo)
{
    static int iFact = 1;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    if(iNo > 1)
    {
        iFact = iFact * iNo;
        Fact(iNo-1);
    }
    return iFact;
}

int main()
{
    int iValue = 0, iRet = 0;

    printf("Enter number : ");
    scanf("%d",&iValue);

    iRet = Fact(iValue);

    printf("%d",iRet);

    return 0;
}

File: Assignment_55/Program4.java
/* Write a java program which accept array of characters from user & Return difference between frequency of capital and frequency of small
characters.
Input :   b N e B R b A I O G i
Output :  3  (7-4)
*/

import java.util.*;

class MyArray
{
    public int Difference(char arr[])
    {
        int i = 0;
        int iCapitalCnt = 0;
        int iSmallCnt = 0;

        for(i = 0;i < arr.length;i++)
        {
            if(arr[i] >= 'A' && arr[i] <= 'Z')
            {
                iCapitalCnt++;
            }
            else if(arr[i] >= 'a' && arr[i] <= 'z')
            {
                iSmallCnt++;
            }
        }
        return (iCapitalCnt - iSmallCnt);
    }
}

class Program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int i = 0;

        System.out.println("Enter characters in the array :");
        String line = sobj.nextLine();
        String[] tokens = line.trim().split("\\s+");
        int size = tokens.length;
        char arr[] = new char[size];
        for (i = 0;i < size;i++)
        {
            arr[i] = arr[i] = tokens[i].charAt(0);
        }

        MyArray mobj = new MyArray();

        int iRet = mobj.Difference(arr);

        System.out.println("Difference is : "+iRet);

        sobj.close();
    }
}